pkg/scheduler/scheduler.go

```diff
@@ -383,8 +386,14 @@ func fits(cq *cache.ClusterQueueSnapshot, usage *workload.Usage, preemptedWorklo
 
 // resourcesToReserve calculates how much of the available resources in cq/cohort assignment should be reserved.
 func resourcesToReserve(e *entry, cq *cache.ClusterQueueSnapshot) workload.Usage {
+	if features.Enabled(features.TopologyAwareScheduling) && e.IsRequestingTAS() {
+		tasRequests := e.assignment.WorkloadsTopologyRequests(&e.Info, cq)
+		tasResult := cq.FindTopologyAssignmentsForWorkload(tasRequests, true)
```

Comment:
Why do we want to simulate empty snapshot here?

Reply:
This is the scenario then the mode is Preempt but there are no targets, for example, because the running workloads are of higher priority. Still we need to reserve the quota and domains so that other workloads don't use it. Simulating empty is the best bet I think. I will add a comment.

Reply:
Add the comment. PTAL. Also we already have a test case "preempting workload with targets reserves capacity so that lower priority workload cannot use it" demonstrating the need for reserving the resources.